Classes/Controller/ProfileController.php

```diff
@@ -53,6 +53,8 @@ public function initializeListAction(): void
         $propertyMappingConfiguration->skipUnknownProperties();
 
         $this->request = $this->request->withArgument('demand', $demandArray);
+
+        $this->settings['showFields'] = !is_array($this->settings['showFields']) ? GeneralUtility::trimExplode(',', $this->settings['showFields']) : $this->settings['showFields'];
```

Comment:
This needs to deal with the fact that `$this->settings['showFields']` 

* may not exists at all in the array
* may not be a string or array (how should this be an array directly ?)

This should be hardened, making it in multiple checks with a logical order of checks and handling.